wordpress-mobile/WordPress-iOS
# Issue/5583 replace themes searchbar

Fixes #5539

This PR resolves the blank space in themes that used to hold both search and the drop down for selecting free/premium/all. The dropdown was hidden and no longer seen/available leaving an empty search area which has now been moved to a search icon button item in the top-right. Additionally, a few methods were renamed to account for the space being about filtering rather than searching.

Note: The original space and dropdown is available in code, in case we ever bring it back, but hardcoded as disabled and without the search button.

To test:

1. Open Themes on a .com site with admin rights.
2. Hit the search icon in the top right, the search bar should display as normal, in the header.
3. 🏁 

Needs review: @nheagy can you take a look?
